yearn/apy/curve/simple.py

```diff
@@ -154,80 +165,160 @@ def simple(vault, samples: ApySamples) -> Apy:
     composite = {
         "boost": boost,
         "pool_apy": pool_apy,
         "boosted_apr": crv_apr,
         "base_apr": base_apr,
-        "cvx_apr": cvx_apr,
+        "cvx_apr": cvx_apy_data.cvx_apr,
         "rewards_apr": reward_apr,
     }
 
     return Apy("crv", gross_apr, net_apy, fees, composite=composite)
+
+class _ConvexVault:
+    def __init__(self, cvx_strategy, vault, gauge, block=None) -> None:
+        self._cvx_strategy = cvx_strategy
+        self.block = block
+        self.vault = vault
+        self.gauge = gauge
+
+    @staticmethod
+    def is_convex_vault(vault) -> bool:
+        """Determines whether the passed in vault is a Convex vault
+        i.e. it only has one strategy that's based on farming Convex.
+        """
+        # prevent circular import for partners calculations
+        from yearn.v2.vaults import Vault as VaultV2
+        if not isinstance(vault, VaultV2):
+            return False 
+
+        return len(vault.strategies) == 1 and "convex" in vault.strategies[0].name.lower()
+
+    def apy(self, base_asset_price, pool_price, base_apr, pool_apy: float, management_fee: float, performance_fee: float) -> Apy:
+        """The standard APY data."""
+        apy_data = self.get_detailed_apy_data(base_asset_price, pool_price, base_apr)
+        gross_apr = (1 + (apy_data.cvx_apr * apy_data.cvx_debt_ratio)) * (1 + pool_apy) - 1
+
+        cvx_net_apr = (apy_data.cvx_apr_minus_keep_crv + apy_data.convex_reward_apr) * (1 - performance_fee) - management_fee
+        cvx_net_farmed_apy = (1 + (cvx_net_apr / COMPOUNDING)) ** COMPOUNDING - 1
+        cvx_net_apy = ((1 + cvx_net_farmed_apy) * (1 + pool_apy)) - 1
+
+        fees = ApyFees(performance=performance_fee, management=management_fee, cvx_keep_crv=apy_data.cvx_keep_crv)
+        return Apy("convex", gross_apr, cvx_net_apy, fees)
+
+    def get_detailed_apy_data(self, base_asset_price, pool_price, base_apr) -> ConvexDetailedApyData:
+        """Detailed data about the apy."""
+        cvx_keep_crv = self._cvx_strategy.keepCRV(block_identifier=self.block) / 1e4
+
+        cvx_booster = contract(addresses[chain.id]['convex_booster'])
+        cvx_fee = self._get_convex_fee(cvx_booster, self.block)
+        convex_reward_apr = self._get_reward_apr(self._cvx_strategy, cvx_booster, base_asset_price, pool_price, self.block)
+
+        cvx_boost = self._get_cvx_boost()
+        cvx_printed_as_crv = self._get_cvx_emissions_printed_as_crv()
+        cvx_apr = ((1 - cvx_fee) * cvx_boost * base_apr) * (1 + cvx_printed_as_crv) + convex_reward_apr
+        cvx_apr_minus_keep_crv = ((1 - cvx_fee) * cvx_boost * base_apr) * ((1 - cvx_keep_crv) + cvx_printed_as_crv)
+        
+        return ConvexDetailedApyData(cvx_apr, cvx_apr_minus_keep_crv, cvx_keep_crv, self._debt_ratio, convex_reward_apr)
+
+    def _get_cvx_emissions_printed_as_crv(self) -> float:
+        """The amount of CVX emissions at the current block, in terms of CRV."""
```

Comment:
"The amount of CVX emissions at the current block for a given pool, converted to CRV (from a pricing standpoint) to ease calculation of total APY."